test/libsolidity/ASTImportTest.cpp

```diff
@@ -0,0 +1,215 @@
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/ASTImportTest.h>
+#include <test/Options.h>
+#include <test/Metadata.h>
+#include <libdevcore/AnsiColorized.h>
+#include <liblangutil/SourceReferenceFormatterHuman.h>
+#include <libsolidity/ast/ASTJsonConverter.h>
+#include <libsolidity/ast/ASTJsonImporter.h>
+#include <libsolidity/interface/CompilerStack.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/throw_exception.hpp>
+#include <fstream>
+#include <memory>
+#include <stdexcept>
+
+
+using namespace langutil;
+using namespace dev::solidity;
+using namespace dev::solidity::test;
+using namespace dev::formatting;
+using namespace dev::test;
+using namespace dev;
+using namespace std;
+namespace fs = boost::filesystem;
+using namespace boost::unit_test;
+
+ASTImportTest::ASTImportTest(string const& _filename)
+{
+	// TODO figure out where this is given a .sol file as input and change it to be the JSON
+	if (!boost::algorithm::ends_with(_filename, ".sol"))
+		BOOST_THROW_EXCEPTION(runtime_error("Invalid test contract file name: \"" + _filename + "\"."));
+
+	// use .json as input file for the test
+	m_astFilename = _filename.substr(0, _filename.size() - 4) + ".json";
+
+	// try parsing as json
+	Json::Value* ast = new Json::Value();
+	// meanwhile set sourceIndices map so that ASTJsonConverter knows which index to give in SourceLocations
+	size_t i = 0;
+	if (jsonParseFile(m_astFilename, *ast))
+	{
+		if ((*ast).isMember("sourceList") && (*ast).isMember("sources"))
+		{
+			for (auto& src: (*ast)["sourceList"])
+			{
+				astAssert( (*ast)["sources"][src.asString()]["AST"]["nodeType"].asString() == "SourceUnit",  "Top-level node should be a 'SourceUnit'");
+				m_sourceJsons[src.asString()] = &((*ast)["sources"][src.asString()]["AST"]);
+				m_sourceIndices[src.asString()] = i;
+				i++;
+			}
+		}
+		else
+		{
+			BOOST_THROW_EXCEPTION(InvalidAstError() << errinfo_comment("should have been option A")); //TODO make this helpful
+		}
+	}
+	else
+	{
+		BOOST_THROW_EXCEPTION(InvalidAstError() << errinfo_comment("Input file could not be parsed to JSON"));
+	}
+
+	// save entire Json from input as string in m_expectations
+	// TODO but modify the metadata to not includes the hash of the solidity-source code
+	// TODO2: byteCodeSansMetadata might need to be modified to remove metadata of imported contracts too
+	if ( (*ast).isMember("contracts") )
+	{
+		for (auto& srcName: (*ast)["contracts"].getMemberNames())
+		{
+			// why is this not importing correctly? HELP namesspace issues?
+//			string bin = (*ast)["contracts"][srcName]["bin"].asString();
+//			string binSansMetadata = dev::test::bytecodeSansMetadata(bin);
```

Comment:
somehow this is not working.

my editor finds it, but when i compile it does not :/

![image](https://user-images.githubusercontent.com/9882716/62200839-7d425700-b386-11e9-8d39-9ef448930b4c.png)

Reply:
solved by adding ```../Metadata.cpp``` to /test/tools/CMakelists.txt